ui/app/registration/controllers/visitController.js

```diff
@@ -229,7 +229,8 @@ angular.module('bahmni.registration')
             };
             var isValid = function (mandatoryConcepts) {
                 var concept = mandatoryConcepts.filter(function (mandatoryConcept) {
-                    if (mandatoryConcept.isNumeric() && mandatoryConcept.value === 0) {
+                    if ((mandatoryConcept.isNumeric() && mandatoryConcept.value === 0) ||
+                        (mandatoryConcept.isBoolean && mandatoryConcept.value != null)) {
```

Comment:
Hi @Pasayadaan, 
as you can see [here](https://github.com/Bahmni/openmrs-module-bahmniapps/blob/master/ui/app/common/concept-set/models/conceptSetObservation.js#L325-L339), `hasValue` method already takes care of all the checks. So, I feel if you just use `mandatoryConcept.hasValue()` method, it will take care of everything.